### PR TITLE
[6014] Delete dialog - When users checks the Dependent Items the Child Items should update the child items to remove

### PR DESCRIPTION
**Pull request by jvega190**
https://github.com/craftercms/craftercms/issues/6014

### CODE REVIEW

File: ui/app/src/components/DeleteDialog/DeleteDialogContainer.tsx [hunk 1]
```diff
@@ -125,16 +130,18 @@ export function DeleteDialogContainer(props: DeleteDialogContainerProps) {
   };
 
   const onSelectAllDependantClicked = () => {
-    const setChecked = Boolean(dependentItems.find((path) => !selectedItems[path]));
+    const selectableDependent = dependentItems.filter((item) => !disabledDependentItems.includes(item));
+    const setChecked = Boolean(selectableDependent.find((path) => !selectedItems[path]));
```
Comment (rart) [inline ui/app/src/components/DeleteDialog/DeleteDialogContainer.tsx]: Perhaps .some is more appropriate here

File: ui/app/src/components/DeleteDialog/DeleteDialogContainer.tsx [hunk 2]
```diff
@@ -59,6 +61,9 @@ export function DeleteDialogContainer(props: DeleteDialogContainerProps) {
   const [submitDisabled, setSubmitDisabled] = useState(true);
   const [confirmChecked, setConfirmChecked] = useState(false);
   const authoringBase = useSelection((state) => state.env.authoringBase);
+  const itemsByPath = useItemsByPath();
+  useFetchSandboxItems(dependentItems ?? []);
```
Comment (rart) [inline ui/app/src/components/DeleteDialog/DeleteDialogContainer.tsx:65]: How are other items fetched? Could we use same block to load these?

File: ui/app/src/components/DependencySelection/SelectionList.tsx [hunk 3]
```diff
@@ -105,6 +106,7 @@ export function SelectionList(props: SelectionListProps) {
     onSelectAllClicked,
     selectedItems,
     disabled = false,
+    disabledPaths = [],
```
Comment (rart) [inline ui/app/src/components/DependencySelection/SelectionList.tsx]: Might perform better for this to be a lookup?

**Review by rart — CHANGES_REQUESTED** [review]

File: ui/app/src/components/DeleteDialog/DeleteDialogContainer.tsx (hunk 2 above)
Comment (jvega190) [reply ui/app/src/components/DeleteDialog/DeleteDialogContainer.tsx:65]: The items are a prop of the DeleteDialog, so they're coming from outside the dialog (the items selected for deletion)

File: ui/app/src/components/DeleteDialog/DeleteDialogContainer.tsx [hunk 4]
```diff
@@ -59,6 +61,12 @@ export function DeleteDialogContainer(props: DeleteDialogContainerProps) {
...
+  const disabledDependentItems = createPresenceTable(
+    dependentItems?.filter((item) => !itemsByPath[item].availableActionsMap.delete) ?? [],
```
Comment (rart) [inline ui/app/src/components/DeleteDialog/DeleteDialogContainer.tsx]: Specially with the new logic, is there a moment where `itemsByPath` does't have all the paths loaded? You should throttle for testing
```suggestion
    dependentItems?.filter((path) => !itemsByPath[path]?.availableActionsMap.delete) ?? [],
```

**Review by rart — CHANGES_REQUESTED** [review]

**Review by rart — APPROVED** [review]

**Pull request merged**